src/Apps/MyCollection/Routes/MyCollectionArtwork/Components/MyCollectionArtworkSWASection.tsx

```diff
@@ -1,149 +1,134 @@
 
-  return (
-    <>
-      <Text mb={0.5} mt={2} variant="sm-display">
-        Interested in Selling This Work?
-      </Text>
-      <Text mb={2} color="black60" variant="xs">
-        Let our experts find the best sales option for you.
-      </Text>
-      <RouterLink
-        textDecoration="none"
-        to={route}
-        data-testid="submit-for-sale-link"
-      >
-        <Button
-          variant="primaryBlack"
-          width="100%"
-          mb={2}
-          data-testid="submit-for-sale"
-          onClick={() => {
-            tracking.trackEvent({
-              contextModule: ContextModule.topWorksRail,
-              context_page_owner_type: OwnerType.myCollectionArtwork,
-              context_page_owner_id: artworkId,
-              context_page_owner_slug: slug,
-              destination_page_owner_type: OwnerType.consignmentFlow,
-              subject: "Submit This Artwork to Sell",
-              platform: "web",
-            })
-          }}
-        >
-          Submit for Sale
-        </Button>
-      </RouterLink>
+  const handleSubmitArtwork = async () => {
+    tracking.trackEvent({
+      contextModule: ContextModule.topWorksRail,
+      context_page_owner_type: OwnerType.myCollectionArtwork,
+      context_page_owner_id: artwork.internalID,
+      context_page_owner_slug: artwork?.artist?.slug,
+      destination_page_owner_type: OwnerType.consignmentFlow,
+      subject: "Submit This Artwork to Sell",
+      platform: "web",
+    })
 
-      <Text color="black60" variant="xs">
-        Learn more about{" "}
-        <Clickable
-          onClick={() => {
-            learnMore()
-            tracking.trackEvent({
-              contextModule: ContextModule.topWorksRail,
-              context_page_owner_type: OwnerType.myCollectionArtwork,
-              context_page_owner_id: artworkId,
-              context_page_owner_slug: slug,
-              subject: "Learn More",
-              platform: "web",
-            })
-          }}
-          color="black60"
-          textDecoration="underline"
-          data-testid="learn-more"
-        >
-          selling with Artsy.
-        </Clickable>
-      </Text>
-      <Separator my={2} />
-    </>
-  )
-}
+    if (!enableNewSubmissionFlow) return
 
-export const MyCollectionArtworkSWASectionDesktopLayout: React.FC<Props> = ({
-  route,
-  learnMore,
-  slug,
-  artworkId,
-  ctaColor = "primaryBlack",
-}) => {
-  const tracking = useTracking()
+    setIsLoading(true)
+
+    try {
+      let submissionID = artwork.consignmentSubmission?.internalID
+
+      if (!submissionID) {
+        submissionID = await createOrUpdateConsignSubmission(relayEnvironment, {
+          sessionID: !isLoggedIn ? getENV("SESSION_ID") : undefined,
+          artistID: artwork.artist?.internalID,
+          myCollectionArtworkID: artwork.internalID,
+          // Source is necessary in order to link this to a mycollection artwork
+          source: "MY_COLLECTION",
```

Comment:
this can be removed now

Reply:
I suppose we do not have to set `source: "MY_COLLECTION"` anymore whenever `myCollectionArtworkID` is set when submitting an artwork from My Collection, right?

Reply:
Will `source` be automatically set as well when updating a submission?

Reply:
https://github.com/artsy/force/pull/14071

Reply:
You are right, source is set automatically only in `createConsignmentSubmission` mutation: https://github.com/artsy/metaphysics/pull/5803/files#diff-d5fd9257b7014a414628f2481fe406669003b05c3b5d0a8113cb6e67137bc7b6R129